File: two-pointers/valid-palindrome/valid-palindrome.cs
public class Solution
{
    public bool IsPalindrome(string s)
    {
        if (string.IsNullOrEmpty(s))
        {
            return false;
        }

        int left = 0;
        int right = s.Length - 1;

        while (left <= right)
        {
            if (!char.IsLetterOrDigit(s[left]))
            {
                left++;
                continue;
            }

            if (!char.IsLetterOrDigit(s[right]))
            {
                right--;
                continue;
            }

            if (char.ToLower(s[left]) != char.ToLower(s[right]))
            {
                return false;
            }

            right--;
            left++;
        }

        return true;
    }
}

File: two-pointers/valid-palindrome/valid-palindrome.go
func isPalindrome(s string) bool {
    if len(s) == 0 {
        return false
    }

    left, right := 0, len(s)-1

    for left < right {
		lRune := rune(s[left])
        if (!unicode.IsLetter(lRune) && !unicode.IsDigit(lRune)) {
            left++
			continue
        }

		rRune := rune(s[right])
        if (!unicode.IsLetter(rRune) && !unicode.IsDigit(rRune)) {
            right--
			continue
        }

        if unicode.ToLower(lRune) != unicode.ToLower(rRune) {
            return false
        }

        left++
        right--
    }

    return true
}